src/main/java/database/DatabaseController.java

```diff
@@ -182,6 +274,67 @@ public ArrayList<Listing> getAllListings() throws IOException {
 
     }
 
+    /**
+     * adds a listing to the cart based on the ID of the listing passed
+     *
+     * @param ID pass the id of the listing to be removed
+     */
+    @Override
+    public void addToCart(int ID) throws IOException {
+        User currUser = Main.getCurrentUser();
+        Cart currCart = currUser.getCart();
+        Listing listing = getListingByID(ID);
+        currCart.addItem(listing);
+    }
+
+    /**
+     * removes a listing from the cart based on the ID of the listing passed
+     *
+     * @param ID pass the id of the listing to be removed
+     */
+    @Override
+    public void removeFromCart(int ID) throws IOException {
+        User currUser = Main.getCurrentUser();
+        Cart currCart = currUser.getCart();
+        Listing listing = getListingByID(ID);
+        currCart.removeItem(listing);
+    }
+
+    /**
+     * adds a review rating to a given user
+     *
+     * @param reviewed user being reviewed
+     * @param rating number given by the reviewer
+     */
+    @Override
+    public void addReview(User reviewed, int rating) throws IOException {
+        User reviewedUser = getUserWithUsername(reviewed.getUsername());
+        ArrayList<Integer> reviewedUserRatings = reviewedUser.getReviews();
+        reviewedUserRatings.add(rating);
+    }
```

Comment:
this doesn't work. you need to then actually make the change in the database, or the changes will not persist